diff --git a/src/nautilus-enums.h b/src/nautilus-enums.h
index 022e62251..f3c5a403e 100644
--- a/src/nautilus-enums.h
+++ b/src/nautilus-enums.h
@@ -33,7 +33,7 @@ typedef enum
     NAUTILUS_GRID_ICON_SIZE_SMALL       = 48,
     NAUTILUS_GRID_ICON_SIZE_SMALL_PLUS  = 64,
     NAUTILUS_GRID_ICON_SIZE_MEDIUM      = 96,
-    NAUTILUS_GRID_ICON_SIZE_LARGE       = 168,
+    NAUTILUS_GRID_ICON_SIZE_LARGE       = 128,
     NAUTILUS_GRID_ICON_SIZE_EXTRA_LARGE = 256,
 } NautilusGridIconSize;
 
