diff --git a/src/color_dialog/src/main.ts b/src/color_dialog/src/main.ts
index 9522499..16968b9 100644
--- a/src/color_dialog/src/main.ts
+++ b/src/color_dialog/src/main.ts
@@ -1,4 +1,4 @@
-#!/usr/bin/gjs
+#!/usr/bin/env gjs
 
 imports.gi.versions.Gtk = '3.0';
 
@@ -6,7 +6,7 @@ const { Gio, GLib, Gtk, Gdk } = imports.gi;
 
 const EXT_PATH_DEFAULTS = [
     GLib.get_home_dir() + "/.local/share/gnome-shell/extensions/",
-    "/usr/share/gnome-shell/extensions/"
+    "/run/current-system/sw/share/gnome-shell/extensions/"
 ];
 const DEFAULT_HINT_COLOR = 'rgba(251, 184, 108, 1)'; //pop-orange
 
@@ -84,4 +84,4 @@ function launch_color_dialog() {
 
 Gtk.init(null);
 
-launch_color_dialog()
\ No newline at end of file
+launch_color_dialog()
diff --git a/src/dialog_launcher.ts b/src/dialog_launcher.ts
index 2039bef..3e53e00 100644
--- a/src/dialog_launcher.ts
+++ b/src/dialog_launcher.ts
@@ -25,8 +25,8 @@ const DATA_DIRS: string = GLib.get_system_data_dirs();
 /// Search paths for finding applications
 const SEARCH_PATHS: Array<[string, string]> = [
     // System-wide
-    ["System", "/usr/share/applications/"],
-    ["System (Local)", "/usr/local/share/applications/"],
+    ["System", "/run/current-system/sw/share/applications/"],
+    ["System (Local)", "/run/current-system/sw/local/share/applications/"],
     // User-local
     ["User", HOME_DIR + "/.local/share/applications/"],
     // System-wide flatpaks
diff --git a/src/extension.ts b/src/extension.ts
index 29c89eb..39fe4e6 100644
--- a/src/extension.ts
+++ b/src/extension.ts
@@ -503,7 +503,7 @@ export class Ext extends Ecs.System<ExtEvent> {
     exception_dialog() {
         let path = Me.dir.get_path() + "/floating_exceptions/main.js";
 
-        utils.async_process(["gjs", path], null, null)
+        utils.async_process([path], null, null)
             .then(output => {
                 log.debug(`Floating Window Dialog Event: ${output}`)
                 switch (output.trim()) {
diff --git a/src/floating_exceptions/src/main.ts b/src/floating_exceptions/src/main.ts
index 719c29b..7423d95 100644
--- a/src/floating_exceptions/src/main.ts
+++ b/src/floating_exceptions/src/main.ts
@@ -1,4 +1,4 @@
-#!/usr/bin/gjs
+#!/usr/bin/env gjs
 
 imports.gi.versions.Gtk = '3.0'
 
diff --git a/src/launcher_service.ts b/src/launcher_service.ts
index 7a12bb4..ca7c754 100644
--- a/src/launcher_service.ts
+++ b/src/launcher_service.ts
@@ -52,7 +52,7 @@ export var BUILTINS: Array<PluginType.Source> = [
 const LOCAL_PLUGINS: string = GLib.get_home_dir() + "/.local/share/pop-shell/launcher/"
 
 /** Launcher plugins that are installed system-wide */
-const SYSTEM_PLUGINS: string = "/usr/lib/pop-shell/launcher/"
+const SYSTEM_PLUGINS: string = "/run/current-system/sw/lib/pop-shell/launcher/"
 
 export class LauncherService {
     private plugins: Map<string, PluginType.Source> = new Map()
diff --git a/src/panel_settings.ts b/src/panel_settings.ts
index a2e9662..46c2136 100644
--- a/src/panel_settings.ts
+++ b/src/panel_settings.ts
@@ -332,7 +332,7 @@ function color_selector(ext: Ext, menu: any) {
     color_selector_item.add_child(color_button);
     color_button.connect('button-press-event', () => {
         let path = Me.dir.get_path() + "/color_dialog/main.js";
-        let resp = GLib.spawn_command_line_async(`gjs ${path}`);
+        let resp = GLib.spawn_command_line_async(`${path}`);
         if (!resp) {
 
             return null;
@@ -347,4 +347,4 @@ function color_selector(ext: Ext, menu: any) {
     });
 
     return color_selector_item;
-}
\ No newline at end of file
+}
diff --git a/src/plugin_scripts.ts b/src/plugin_scripts.ts
index 94c874a..377000b 100644
--- a/src/plugin_scripts.ts
+++ b/src/plugin_scripts.ts
@@ -12,7 +12,7 @@ import type { Response } from 'launcher_plugins'
 const LOCAL: string = GLib.get_home_dir() + "/.local/share/pop-shell/scripts/"
 
 /** Scripts maintained by this project, or the distribution */
-const SYSTEM: string = "/usr/lib/pop-shell/scripts/"
+const SYSTEM: string = "/run/current-system/sw/lib/pop-shell/scripts/"
 
 interface ScriptData {
     name: string
@@ -142,4 +142,4 @@ export class ScriptsBuiltin extends plugins.Builtin {
 
         return { event: "noop" }
     }
-}
\ No newline at end of file
+}
diff --git a/src/plugins/calc/main.js b/src/plugins/calc/main.js
index 3139f6d..713a38f 100755
--- a/src/plugins/calc/main.js
+++ b/src/plugins/calc/main.js
@@ -1,4 +1,4 @@
-#!/usr/bin/gjs
+#!/usr/bin/env gjs
 
 const { GLib, Gio } = imports.gi;
 
diff --git a/src/plugins/files/main.js b/src/plugins/files/main.js
index ba1a210..efa94e5 100755
--- a/src/plugins/files/main.js
+++ b/src/plugins/files/main.js
@@ -1,4 +1,4 @@
-#!/usr/bin/gjs
+#!/usr/bin/env gjs
 
 const { GLib, Gio } = imports.gi;
 
diff --git a/src/plugins/pulse/main.js b/src/plugins/pulse/main.js
index 1392ca6..df2e6d8 100755
--- a/src/plugins/pulse/main.js
+++ b/src/plugins/pulse/main.js
@@ -1,4 +1,4 @@
-#!/usr/bin/gjs
+#!/usr/bin/env gjs
 
 const { GLib, Gio } = imports.gi;
 
diff --git a/src/plugins/recent/main.js b/src/plugins/recent/main.js
index 91ad102..288f19d 100755
--- a/src/plugins/recent/main.js
+++ b/src/plugins/recent/main.js
@@ -1,4 +1,4 @@
-#!/usr/bin/gjs
+#!/usr/bin/env gjs
 
 const { GLib, Gio, Gtk } = imports.gi
 
diff --git a/src/plugins/terminal/main.js b/src/plugins/terminal/main.js
index 7446340..0564443 100755
--- a/src/plugins/terminal/main.js
+++ b/src/plugins/terminal/main.js
@@ -1,4 +1,4 @@
-#!/usr/bin/gjs
+#!/usr/bin/env gjs
 
 const { GLib, Gio } = imports.gi;
 
