diff --git a/interfaces/system_key.go b/interfaces/system_key.go
index a8981290a1..f9085e32bb 100644
--- a/interfaces/system_key.go
+++ b/interfaces/system_key.go
@@ -251,7 +251,7 @@ func SystemKeyMismatch() (bool, error) {
 		if exe, err := os.Readlink("/proc/self/exe"); err == nil {
 			// detect running local local builds
 			if !strings.HasPrefix(exe, "/usr") && !strings.HasPrefix(exe, "/snap") {
-				logger.Noticef("running from non-installed location %s: ignoring system-key", exe)
+				logger.Debugf("running from non-installed location %s: ignoring system-key", exe)
 				return false, ErrSystemKeyVersion
 			}
 		}
