diff --git a/src/libexpr/flake/flakeref.cc b/src/libexpr/flake/flakeref.cc
index 833e8a776..54bca958f 100644
--- a/src/libexpr/flake/flakeref.cc
+++ b/src/libexpr/flake/flakeref.cc
@@ -155,8 +155,11 @@ std::pair<FlakeRef, std::string> parseFlakeRefWithFragment(
             }
 
         } else {
-            if (!hasPrefix(path, "/"))
-                throw BadURL("flake reference '%s' is not an absolute path", url);
+            if (!hasPrefix(path, "/")) {
+                path = absPath(path);
+                if (!hasPrefix(path, absPath(".")))
+                    throw BadURL("flake reference '%s' is not an absolute or sub-path", url);
+            }
             auto query = decodeQuery(match[2]);
             path = canonPath(path + "/" + get(query, "dir").value_or(""));
         }
