diff --git a/src/cmd/create.go b/src/cmd/create.go
index c50460a..defd8a4 100644
--- a/cmd/create.go
+++ b/cmd/create.go
@@ -57,6 +57,7 @@ var (
 	}{
 		{"/etc/profile.d/toolbox.sh", "/etc/profile.d/toolbox.sh"},
 		{"/etc/profile.d/toolbox.sh", "/usr/share/profile.d/toolbox.sh"},
+		{"/etc/profile.d/toolbox.sh", "/run/current-system/sw/share/profile.d/toolbox.sh"},
 	}
 )

@@ -347,6 +348,18 @@ func createContainer(container, image, release, authFile string, showCommandToEn
 		runMediaMount = []string{"--volume", "/run/media:/run/media:rslave"}
 	}

+	var nixStoreMount []string
+
+	if utils.PathExists("/nix/store") {
+		nixStoreMount = []string{"--volume", "/nix/store:/nix/store:ro"}
+	}
+
+	var runCurrentSystemMount []string
+
+	if utils.PathExists("/run/current-system") {
+		runCurrentSystemMount = []string{"--volume", "/run/current-system:/run/current-system:ro"}
+	}
+
 	logrus.Debug("Looking up toolbox.sh")

 	var toolboxShMount []string
@@ -434,6 +447,8 @@ func createContainer(container, image, release, authFile string, showCommandToEn
 	createArgs = append(createArgs, mntMount...)
 	createArgs = append(createArgs, pcscSocketMount...)
 	createArgs = append(createArgs, runMediaMount...)
+	createArgs = append(createArgs, nixStoreMount...)
+	createArgs = append(createArgs, runCurrentSystemMount...)
 	createArgs = append(createArgs, toolboxShMount...)

 	createArgs = append(createArgs, []string{
